WireDomain/Tests/WireDomainTests/Repositories/ConversationRepositoryTests.swift

```diff
@@ -295,34 +295,102 @@ final class ConversationRepositoryTests: XCTestCase {
 
             let conversation = try XCTUnwrap(ZMConversation.fetch(with: Scaffolding.conversationID, in: context), "No Conversation")
 
-            try checkLastMessage(
+            try internalTest_checkLastMessage(
                 in: teamConversation,
-                isLeaveMessageFor: user,
+                messageType: .teamMemberLeave,
                 at: timestamp
             )
-
-            try checkLastMessage(
+            
+            try internalTest_checkLastMessage(
                 in: teamAnotherConversation,
-                isLeaveMessageFor: user,
+                messageType: .teamMemberLeave,
                 at: timestamp
             )
 
             let lastMessage = try XCTUnwrap(conversation.lastMessage as? ZMSystemMessage)
             XCTAssertNotEqual(lastMessage.systemMessageType, .teamMemberLeave, "Should not append leave message to regular conversation")
         }
     }
+    
+    
+    func testFetchConversation_It_Retrieves_Conversation_Locally() async {
+        // Given
 
-    private func checkLastMessage(
-        in conversation: ZMConversation,
-        isLeaveMessageFor user: ZMUser,
-        at timestamp: Date
-    ) throws {
-        let lastMessage = try XCTUnwrap(conversation.lastMessage as? ZMSystemMessage, "Last message is not system message")
+        let conversation = await context.perform { [self] in
+            modelHelper.createGroupConversation(
+                id: Scaffolding.conversationID,
+                domain: Scaffolding.domain,
+                in: context
+            )
+        }
+
+        // When
 
-        XCTAssertEqual(lastMessage.systemMessageType, .teamMemberLeave, "System message is not teamMemberLeave: but '\(lastMessage.systemMessageType.rawValue)")
+        let localConversation = await sut.fetchConversation(
+            with: Scaffolding.conversationID,
+            domain: Scaffolding.domain
+        )
```

Comment:
Note regarding testing our repositories: for now we don't inject mocked local stores, we indirectly test that storage logic is correctly implemented but when all our local stores are in place, I think we should be injecting mock local stores in the repo and create UTs for all our stores to test our storage logic directly.

Reply:
Agreed